plugin/torch/torch_criterion-inl.h

```diff
@@ -61,7 +60,7 @@ class TorchCriterionOp : public Operator {
     int err = lua_pcall(L, 0, 1, 0);
     CHECK_EQ(err, 0) << lua_tostring(L, -1);
     // serialize
-    TorchState::Serialize(&chunk_);
+    this->lua_reference_ = lua_ref(L, LUA_REGISTRYINDEX);
```

Comment:
We could do that. I left it with references as I think it is a lot cleaner and easier to ensure the lua stack is entirely empty at the beginning of functions. I can change all the checks to ensure 1 element and remove the reference if you would like.